src/app/pipes/array/unique.ts

```diff
@@ -1,11 +1,19 @@
-import {PipeTransform, Pipe} from '@angular/core';
+import { PipeTransform, Pipe } from '@angular/core';
 
-@Pipe({name: 'unique'})
+@Pipe({ name: 'unique' })
 export class UniquePipe implements PipeTransform {
 
-  transform(input: any): any[] {
-    return Array.isArray(input)
-      ? input.filter((e, i) => input.indexOf(e) === i)
-      : input;
+  transform(input: any, propertyName?: string): any[] {
+    const uniques: boolean[] = [];
+    return Array.isArray(input) ?
+      typeof propertyName === 'undefined' ?
+        input.filter((e, i) => input.indexOf(e) === i) :
+        input.filter((e, i) => {
+          if (typeof e !== 'object' || !e.hasOwnProperty(propertyName) || uniques[e[propertyName]]) {
```

Comment:
@IARayan Thank you for your pull-request! 
We just need a bit more modifications & improvement before we get it into `ngx-pipes`.